src/pybind.cpp

```diff
@@ -7,19 +7,36 @@ namespace py = pybind11;
 PYBIND11_MODULE(hilok, m)
 {   
     m.doc() = "Hierarchical lock manager";
-    py::class_<HiLok>(m, "HiLok")
+    py::class_<HiLok, std::shared_ptr<HiLok>>(m, "HiLok")
         .def(py::init<>())
-        .def("write", &HiLok::write, py::arg("path"), py::arg("block") = true, py::arg("timeout") = 0)
-        .def("read", &HiLok::read, py::arg("path"), py::arg("block") = true, py::arg("timeout") = 0)
+        .def(py::init<char>(), py::arg("sep") = '/')
+        .def(py::init<char, bool>(), py::arg("sep") = '/', py::arg("recursive") = true)
+        .def("write", [](std::shared_ptr<HiLok> lok, std::string_view path, std::optional<bool> block, std::optional<double> timeout) {
+                py::gil_scoped_release();
```

Comment:
This needs to have a name, or it'll be destroyed immediately.

```suggestion
                py::gil_scoped_release _gil_rel;
```

Reply:
done